### PR TITLE
Force feature toggle for tasks to false when unsupported

### PR DESCRIPTION
**Pull request by ericbottard**
Fixes https://github.com/spring-cloud/spring-cloud-dataflow-server-cloudfoundry/issues/297

Depends on https://github.com/spring-cloud/spring-cloud-deployer-cloudfoundry/issues/169

### CODE REVIEW

File: spring-cloud-dataflow-server-cloudfoundry-autoconfig/src/main/java/org/springframework/cloud/dataflow/server/cloudfoundry/config/TaskFeatureAutoToggleEnvironmentPostProcessor.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	@Override
+	public void postProcessEnvironment(ConfigurableEnvironment environment, SpringApplication application) {
+
+		// Create a throwaway AppContext to connect to CC API and assess its version.
+		try {
+			AnnotationConfigApplicationContext applicationContext
+				= new AnnotationConfigApplicationContext();
+			applicationContext.register(CloudFoundryDeployerAutoConfiguration.EarlyConnectionConfiguration.class);
+			applicationContext.setEnvironment(environment); // Inherit current environment
+			applicationContext.refresh();
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-server-cloudfoundry-autoconfig/src/main/java/org/springframework/cloud/dataflow/server/cloudfoundry/config/TaskFeatureAutoToggleEnvironmentPostProcessor.java]: Should this `applicationContext` be closed at the end of this method?

**Comment by ericbottard** [comment]
Good call

File: spring-cloud-dataflow-server-cloudfoundry-autoconfig/src/main/java/org/springframework/cloud/dataflow/server/cloudfoundry/config/TaskFeatureAutoToggleEnvironmentPostProcessor.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+	private static final String TASKS_KEY = FeaturesProperties.FEATURES_PREFIX + "." + FeaturesProperties.TASKS_ENABLED;
+
+	private static final Logger logger = LoggerFactory.getLogger(TaskFeatureAutoToggleEnvironmentPostProcessor.class);
+
+	@Override
+	public void postProcessEnvironment(ConfigurableEnvironment environment, SpringApplication application) {
+
+		// Create a throwaway AppContext to connect to CC API and assess its version.
+		try {
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-server-cloudfoundry-autoconfig/src/main/java/org/springframework/cloud/dataflow/server/cloudfoundry/config/TaskFeatureAutoToggleEnvironmentPostProcessor.java]: We also need to protect against multiple invocation of this EnvironmentPostProcessor.

Comment (ilayaperumalg) [inline spring-cloud-dataflow-server-cloudfoundry-autoconfig/src/main/java/org/springframework/cloud/dataflow/server/cloudfoundry/config/TaskFeatureAutoToggleEnvironmentPostProcessor.java]: Will fix while merging.

Comment (ilayaperumalg) [inline spring-cloud-dataflow-server-cloudfoundry-autoconfig/src/main/java/org/springframework/cloud/dataflow/server/cloudfoundry/config/TaskFeatureAutoToggleEnvironmentPostProcessor.java]: It turns out that when multiple contexts inside the applications invoke this environment post processor, it gets tricky from which application the post processor gets invoked. Though it doesn't hurt invoking this multiple times, the connection configuration is established that many times and thus the application context open/close.  will create a story for the discussion and tracking.

Comment (ilayaperumalg) [inline spring-cloud-dataflow-server-cloudfoundry-autoconfig/src/main/java/org/springframework/cloud/dataflow/server/cloudfoundry/config/TaskFeatureAutoToggleEnvironmentPostProcessor.java]: Created https://github.com/spring-cloud/spring-cloud-dataflow-server-cloudfoundry/issues/304

**Comment by ilayaperumalg** [comment]
LGTM, also verified by connecting to CF environment with the Data Flow CF server locally. Merging with the changes required to close the temporary application context.

**Comment by ilayaperumalg** [comment]
Merged as 2d303843196dc56c5cedc9bf8a24535c78617ee3 along with 45ddff184a6dac06de5cd5a8d78998eb358d7734

**Pull request closed without merging**